targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -198,6 +200,35 @@ void common_rtc_set_interrupt(uint32_t us_timestamp, uint32_t cc_channel,
     nrf_rtc_cc_set(COMMON_RTC_INSTANCE, cc_channel, RTC_WRAP(compare_value));
     nrf_rtc_event_enable(COMMON_RTC_INSTANCE, int_mask);
 }
+
+void common_rtc_set_interrupt(uint32_t us_timestamp, uint32_t cc_channel,
+                              uint32_t int_mask)
+{
+    uint32_t prev_overflows;
+
+    while (1)
```

Comment:
Such a loop should not be placed here, but in common_rtc_32bit_ticks_get(), since this problematic gluing of hardware ticks with software counted overflows can be called from other places as well; currently it is called from us_ticker_read() and lp_ticker_read().
I'd propose to put it like this (not tested):
```
uint32_t common_rtc_32bit_ticks_get(void)
{
    uint32_t ticks;
    uint32_t prev_overflows;
    do {
        prev_overflows = m_common_rtc_overflows;

        ticks = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
        // The counter used for time measurements is less than 32 bit wide,
        // so its value is complemented with the number of registered overflows
        // of the counter.
        ticks += (m_common_rtc_overflows << RTC_COUNTER_BITS);

        int was_masked = __sd_nvic_irq_disable();
        check_overflow();
        if (!was_masked) {
            __sd_nvic_irq_enable();
        }
    } while (m_common_rtc_overflows != prev_overflows);

    return ticks;
}
```